pluralsight/design-system
# CircularProgress/Tab: Run typechecking as part of npm test

### What You're Solving

- There was a regression in a `.d.ts` file (https://github.com/pluralsight/design-system/pull/997) that made it through because we aren't yet running `tsc` to check types as part of CI. 

### Design Decisions

- I considered adding a separate `typecheck` script to the `package.json`, but ended up adding it to `test` because that's an easy way to hook into the CI pipeline, and we had discussed doing it this way previously. I verified that `test` and `test:watch` still have the correct behavior.

### How to Verify

- Tests run (presumably without flagging any type errors, though you could introduce one to verify the new step)
